arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -3665,4 +3715,34 @@ std::shared_ptr<StorageSnapshot> RocksDBEngine::currentSnapshot() {
   }
 }
 
+void RocksDBEngine::removeEmptyJournalFilesFromArchive() {
+  LOG_TOPIC("50812", DEBUG, Logger::ENGINES)
+      << "scanning WAL archive directory for empty files...";
+
+  std::string archiveDirectory =
+      basics::FileUtils::buildFilename(_dbOptions.wal_dir, "archive");
+
+  WRITE_LOCKER(lock, _walFileLock);
+
+  for (auto const& f : basics::FileUtils::listFiles(archiveDirectory)) {
+    if (!f.ends_with(".log")) {
+      // we only care about .log files in there
+      continue;
+    }
+
+    std::string fn = basics::FileUtils::buildFilename(archiveDirectory, f);
+    int64_t size = TRI_SizeFile(fn.c_str());
+    if (size == 0) {
+      // file size is exactly 0 bytes
+      LOG_TOPIC("e79dd", DEBUG, Logger::ENGINES)
+          << "found empty WAL file in archive at startup: '" << f
+          << "', scheduling this file for later deletion";
+      _prunableWalFiles.emplace(basics::FileUtils::buildFilename("archive", f),
+                                TRI_microtime() + _pruneWaitTime);
+    }
+  }
+
+  _metricsPrunableWalFiles.operator=(_prunableWalFiles.size());
```

Comment:
Why operator=?

Reply:
Replaced with Gauge::store(). (this call and others in this file)